lbrynet/core/utils.py

```diff
@@ -134,8 +134,31 @@ def get_sd_hash(stream_info):
         return None
     if isinstance(stream_info, ClaimDict):
         return stream_info.source_hash
-    return stream_info['stream']['source']['source']
+    path = ['claim', 'value', 'stream', 'source', 'source']
+    result = safe_dict_descend(stream_info, *path)
```

Comment:
Would this be sufficient without introducing a lot of extra code?
```
stream_info.get('claim', {}).get('value', {}).get('stream', {}).get('source', {}).get('source')
```

Reply:
After poking through the codebase it doesn't look like there are a lot of deep-dives like this, so `safe_dict_descend` is probably overkill.  While the `get` chain won't catch the case of e.g. a non-dict value ending up somewhere in the path, I don't think that's terribly likely and even if it was we'll still get a stacktrace.

I've written up a few tests to cover `get_sd_hash` to make up for the tests I'm taking out in this commit :)